```
/__w/kokkos/kokkos/containers/unit_tests/TestDynRankViewTypedefs.cpp:69:51: error: 'specialize' is deprecated [clang-diagnostic-deprecated-declarations]
   69 |   static_assert(std::is_same_v<typename ViewType::specialize, void>);
      |                                                   ^
/__w/kokkos/kokkos/containers/unit_tests/TestDynRankViewTypedefs.cpp:203:10: note: in instantiation of function template specialization '(anonymous namespace)::test_view_typedefs_impl<Kokkos::DynRankView<int>, Kokkos::ViewTraits<int *******>, int, Kokkos::LayoutRight, Kokkos::OpenMP, Kokkos::MemoryTraits<>, Kokkos::OpenMP, int, int &>' requested here
  203 |   return test_view_typedefs_impl<Kokkos::DynRankView<T, ViewArgs...>, Kokkos::ViewTraits<T*******, ViewArgs...>,
      |          ^
/__w/kokkos/kokkos/containers/unit_tests/TestDynRankViewTypedefs.cpp:230:17: note: in instantiation of function template specialization '(anonymous namespace)::test_view_typedefs<Kokkos::LayoutRight, Kokkos::OpenMP, Kokkos::MemoryTraits<>, Kokkos::OpenMP, int, int &, int>' requested here
  230 |   static_assert(test_view_typedefs<layout_type, space, memory_traits, host_mirror_space, int, int&>(
      |                 ^
/__w/kokkos/kokkos/containers/unit_tests/../src/Kokkos_DynRankView.hpp:472:20: note: 'specialize' has been explicitly marked deprecated here
  472 |   using specialize KOKKOS_DEPRECATED = void;
      |                    ^
/__w/kokkos/kokkos/core/src/Kokkos_Macros.hpp:580:29: note: expanded from macro 'KOKKOS_DEPRECATED'
  580 | #define KOKKOS_DEPRECATED [[deprecated]]
      |                             ^

--- a/core/src/Cuda/Kokkos_CudaSpace.cpp
+++ b/core/src/Cuda/Kokkos_CudaSpace.cpp
@@ -178,7 +178,7 @@ void *impl_allocate_common(const int device_id,
                 "Please update your CUDA runtime version or "
                 "reconfigure with "
                 "-D Kokkos_ENABLE_IMPL_CUDA_UNIFIED_MEMORY=OFF");
-  if (arg_alloc_size) {  // cudaMemAdvise_v2 does not work with nullptr
+  if (arg_alloc_size) {  // cudaMemAdvise does not work with nullptr
     error_code = cudaMallocManaged(&ptr, arg_alloc_size, cudaMemAttachGlobal);
     if (error_code == cudaSuccess) {
       // One would think cudaMemLocation{device_id,
@@ -187,8 +187,14 @@ void *impl_allocate_common(const int device_id,
       cudaMemLocation loc;
       loc.id   = device_id;
       loc.type = cudaMemLocationTypeDevice;
+#if CUDART_VERSION < 13000  // cudaMemAdvise_v2 was deprecated in CUDA 13, it is
+                            // now the same as cudaMemAdvise
       KOKKOS_IMPL_CUDA_SAFE_CALL(cudaMemAdvise_v2(
           ptr, arg_alloc_size, cudaMemAdviseSetPreferredLocation, loc));
+#else
+      KOKKOS_IMPL_CUDA_SAFE_CALL(cudaMemAdvise(
+          ptr, arg_alloc_size, cudaMemAdviseSetPreferredLocation, loc));
+#endif
     }
   }
 #elif (defined(KOKKOS_ENABLE_IMPL_CUDA_MALLOC_ASYNC) && CUDART_VERSION >= 11020)

